cmd/osv-scanner/main.go

```diff
@@ -147,13 +157,26 @@ func run(args []string, stdout, stderr io.Writer) int {
 				return fmt.Errorf("failed to write output: %w", errPrint)
 			}
 
+			if outputPath != "" {
```

Comment:
Can you add a comment for why we do this? I.e. we only want to create a file on success?

Reply:
Updated it to now write directly to output file as described here: https://github.com/google/osv-scanner/pull/420#discussion_r1244619219